app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,77 @@
+package agency.five.codebase.android.movieapp.ui.component
+import agency.five.codebase.android.movieapp.R
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val category: MovieCategoryLabelTextViewState
+)
+sealed class MovieCategoryLabelTextViewState{
+    class InputText(val text: String): MovieCategoryLabelTextViewState()
+    class ResourceText(@StringRes val textResource: Int): MovieCategoryLabelTextViewState()
+}
+
+@Composable
+fun MovieCategoryLabel(movieCategory: MovieCategoryLabelViewState)
+{
+    if(movieCategory.isSelected){ Selected(movieCategory) } else{ Unselected(movieCategory) }
+}
+
+@Composable
+fun Selected(movieCategory: MovieCategoryLabelViewState,modifier: Modifier = Modifier)
+{
+    Column(modifier = modifier.width(intrinsicSize = IntrinsicSize.Max))
+    {
+        Text(
+            modifier = modifier.fillMaxWidth(),
+            text = selectTextSource(movieCategory = movieCategory),
+            color = Color.Black,
+            fontWeight = FontWeight.Bold,
+            fontSize = 20.sp
+
+        )
+        Spacer(modifier = modifier.size(3.dp))
+        Divider(color = Color.Black, thickness = 4.dp, modifier = modifier.fillMaxWidth())
+    }
+}
+
+@Composable
+fun Unselected(movieCategoryLabelViewState: MovieCategoryLabelViewState)
+{
+    Text(
+        text = selectTextSource(movieCategory = movieCategoryLabelViewState),
+        color = Color.Gray,
+        fontSize = 20.sp
+    )
+}
+
+@Composable
+fun selectTextSource(movieCategory: MovieCategoryLabelViewState): String{
```

Comment:
Composable function should always start with uppercase and also it should not have return type